move memory reads to self thread

tucucu opened the pull request.

ifBars reviewed (approved):
Great contribution! Not only does it greatly improve performance, but it also cleans/organizes the code way better! Thanks for your contribution! - _Tested & Working 11/5 14:25 MT_

I will leave the PR open for now, so that Noob can take a look at it before merging.

tucucu commented:
I wanted to contribute with something as this was the base for my cheat.

I didn't solve anything, just i moved code from one point to another and tbh I didn't expect such a improve on performance. I reviewed the previous code in order to find if I had deleted some sleep or whatever intentional pause, but no, it just works better.

Playing some casual and dm games I didn't find any new unknown issue.

IMXNOOBX commented:
Hi @tucucu , thank you so much for your contribution. im really so sorry for the delay. I will review the pull request asap but from my phone i can really see that you have cleaned the code quite a lot. Thanks again!
